drivers/dma/dma_stm32.c

```diff
@@ -11,14 +11,77 @@
  *        implemented in dma_stm32_v*.c
  */
 
+#include <soc.h>
+#include <init.h>
+#include <drivers/dma.h>
+#include <drivers/clock_control.h>
+#include <clock_control/stm32_clock_control.h>
+
 #include "dma_stm32.h"
 #include <drivers/dma.h>
 #include <soc.h>
 
 #include <logging/log.h>
 LOG_MODULE_REGISTER(dma_stm32, CONFIG_DMA_LOG_LEVEL);
 
-#include <clock_control/stm32_clock_control.h>
+#if defined(CONFIG_DMAMUX_STM32)
+static void dmamux_stm32_irq_handler(void *arg)
+{
+	struct device *dev = arg;
+	struct dmamux_stm32_data *data = dev->driver_data;
+	const struct dmamux_stm32_config *config = dev->config->config_info;
+	DMAMUX_Channel_TypeDef *dmamux = (DMAMUX_Channel_TypeDef *)(config->base);
+
+	int mux_id;
+
+	/* retrieve bit(s) from the dmamux->CSR SO interrupt */
+	for (mux_id = 0; mux_id < data->max_channels; mux_id++) {
+		if (stm32_dmamux_is_enabled(dmamux, mux_id)
+		    && func_ll_is_active_so[mux_id](dmamux)) {
+		/* check if Sync Overrun interrupt is enabled for this channel */
+			break;
+		}
+	}
+
+	if (mux_id == data->max_channels) {
+		LOG_ERR("Unknown interrupt happened.");
+		return;
+	}
+
+	/* clear this irq */
+	func_ll_clear_so[mux_id](dmamux);
+	/* then callback */
+	data->callback_arg = dmamux;
+
+	/* req gen retrieve bit(s) from the dmamux->RGSR RG0x interrupt */
+	/* check if Trigger Overrun interrupt is enabled for this req gen */
+	/* and clear this irq */
+	if ((LL_DMAMUX_GetRequestID(dmamux, mux_id) == LL_DMAMUX_REQ_GEN_0)
+		&& LL_DMAMUX_IsEnabledIT_RGO(dmamux, LL_DMAMUX_REQ_GEN_0)
+		&& LL_DMAMUX_IsActiveFlag_RGO0(dmamux)) {
+		LL_DMAMUX_ClearFlag_RGO0(dmamux);
+	}
+	if ((LL_DMAMUX_GetRequestID(dmamux, mux_id) == LL_DMAMUX_REQ_GEN_1)
+		&& LL_DMAMUX_IsEnabledIT_RGO(dmamux, LL_DMAMUX_REQ_GEN_1)
+		&& LL_DMAMUX_IsActiveFlag_RGO1(dmamux)) {
+		LL_DMAMUX_ClearFlag_RGO1(dmamux);
+	}
+	if ((LL_DMAMUX_GetRequestID(dmamux, mux_id) == LL_DMAMUX_REQ_GEN_2)
+		&& LL_DMAMUX_IsEnabledIT_RGO(dmamux, LL_DMAMUX_REQ_GEN_2)
+		&& LL_DMAMUX_IsActiveFlag_RGO2(dmamux)) {
+		LL_DMAMUX_ClearFlag_RGO2(dmamux);
+	}
+	if ((LL_DMAMUX_GetRequestID(dmamux, mux_id) == LL_DMAMUX_REQ_GEN_3)
+		&& LL_DMAMUX_IsEnabledIT_RGO(dmamux, LL_DMAMUX_REQ_GEN_3)
+		&& LL_DMAMUX_IsActiveFlag_RGO3(dmamux)) {
+		LL_DMAMUX_ClearFlag_RGO3(dmamux);
+	}
+
+	/* then callback */
+	data->callback_arg = dmamux;
```

Comment:
Already done line 54

Reply:
removed from line 54:
note that we could one callback for each type of error ( Syncr or request generator)